### PR TITLE
refactor: move bbs12381g2pub to pkg/crypto

### PR DESCRIPTION
**Pull request by Baha-sk**
Signed-off-by: Baha Shaaban <baha.shaaban@securekey.com>

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ run:
   tests: true
   build-tags: [""]
   skip-dirs:
-    - pkg/doc/bbs/bbs12381g2pub/internal/kilic/bls12-381
+    - pkg/crypto/bbs12381g2pub/internal/kilic/bls12-381
```
Comment (troyronda) [inline .golangci.yml:14]: would it make sense to have a folder between crypto and bbs12381g2pub

/pkg/crypto/.../bbs12381g2pub

Comment (Baha-sk) [reply .golangci.yml:14]: ok, I'll add `bbs` in between just like the original path

Comment (Baha-sk) [reply .golangci.yml]: done

Comment (Baha-sk) [reply .golangci.yml]: renamed `bbs` to `primitive`

**Review by DRK3 — APPROVED** [review]

**Review by troyronda — APPROVED** [review]

**Pull request merged**